modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/EpisodeDao.kt

```diff
@@ -277,6 +277,20 @@ abstract class EpisodeDao {
     @Query("SELECT * FROM podcast_episodes WHERE last_playback_interaction_date IS NOT NULL AND last_playback_interaction_date > 0 ORDER BY last_playback_interaction_date DESC LIMIT 1000")
     abstract fun observePlaybackHistory(): Flowable<List<PodcastEpisode>>
 
+    @Query(
+        """
+        SELECT podcast_episodes.*
+        FROM podcast_episodes
+        LEFT JOIN podcasts ON podcast_episodes.podcast_id = podcasts.uuid
+        WHERE last_playback_interaction_date IS NOT NULL
+          AND last_playback_interaction_date > 0
+          AND (UPPER(podcast_episodes.title) LIKE UPPER(:query)
+               OR UPPER(podcasts.title) LIKE UPPER(:query))
```

Comment:
It's probably not extremely important but when doing user input searches we should escape some special characters.

```sql
-- this example doesn't handle mixed case like yours with UPPER
WHERE column LIKE '%' || :query || '%'  ESCAPE '\'
```

```kotlin
val results = dao.search(term.escapeLike('\\'))

fun String.escapeLike(escapeChar: Char) = replace("$escapeChar", "$escapeChar$escapeChar")
    .replace("%", "$escapeChar%")
    .replace("_", "${escapeChar}_")
```

Otherwise you won't be able to search using `%` and `_` characters. Which is probably not needed given that we search in episode titles. But I think it's worth mentioning.

You can test with this podcast: https://anchor.fm/s/f68090c0/podcast/rss

Reply:
Nice! Added in 18b5d28, tested in a657423